jupyterhub/auth.py

```diff
@@ -769,6 +779,15 @@ def post_spawn_stop(self, user, spawner):
             self.log.warning("Disabling PAM sessions from now on.")
             self.open_sessions = False
 
+    def normalize_username(self, username):
+        """Round-trip the username to normalize it with PAM
+
+        PAM can accept multiple usernames as the same user, normalize them."""
+        if self.pam_normalize_username:
+            import pwd
+            uid = pwd.getpwnam(username).pw_uid
+            username = pwd.getpwuid(uid).pw_name
+        return super().normalize_username(username)
```

Comment:
Should we skip `super()` in the pam case and make it `else: super()`? This would fix case-sensitivity issues. How does the DOMAIN\name case occur?

Reply:
That's a great idea - done.  I also added the username_map support, it's probably not needed with PAM but I guess can't hurt to be consistent...

In my case the weird usernames of DOMAIN\username and email@domain came about because we linked with university ActiveDirectory, and however the linking was done these turned out to be valid usernames for authentication, but not the canonical names in the filesystem.